src/components/ComputeModal.js

```diff
@@ -664,13 +691,16 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
       const { computeZone, computeRegion } = getRegionInfo(location || defaultLocation, locationType)
       const runtimeConfig = currentRuntimeDetails?.runtimeConfig
       const gpuConfig = runtimeConfig?.gpuConfig
+      const autopauseConfig = currentRuntimeDetails ? { enabled: currentRuntimeDetails?.autopauseEnabled, threshold: currentRuntimeDetails?.autopauseThreshold } : null
       const newSparkMode = Utils.switchCase(runtimeConfig?.cloudService,
         [cloudServices.DATAPROC, () => runtimeConfig.numberOfWorkers === 0 ? 'master' : 'cluster'],
         [cloudServices.GCE, () => false],
         [Utils.DEFAULT, () => false] // for when there's no existing runtime
       )
       const isDataproc = !sparkMode && !runtimeConfig?.diskSize
 
+      const calculateAutopauseThreshold = () => autopauseConfig ? (autopauseConfig.enabled ? autopauseConfig.threshold : 0) : defaultAutoPauseLength
```

Comment:
We could write it a bit more concisely using `optional chaining` and `nullish coalescence` like below unless you feel it looks too complicated.
```suggestion
      const calculateAutopauseThreshold = () => autopauseConfig?.enabled ? (autopauseConfig.threshold ?? 0) : defaultAutoPauseLength
```

Also, do we need for it to be a function? It appears to be evaluated only once per `useOnMount()`.